source/AI.cpp

```diff
@@ -1978,7 +1978,15 @@ void AI::PrepareForHyperspace(Ship &ship, Command &command)
 	bool isJump = (ship.GetCheapestJumpType(ship.GetTargetSystem()).first == Ship::JumpType::JumpDrive);
 
 	Point direction = ship.GetTargetSystem()->Position() - ship.GetSystem()->Position();
-	if(!isJump && scramThreshold)
+	const double squaredDeparture =  isJump ?
+		ship.GetSystem()->JumpDepartureDistance() * ship.GetSystem()->JumpDepartureDistance() :
+		ship.GetSystem()->HyperDepartureDistance() * ship.GetSystem()->HyperDepartureDistance();
+	if(ship.Position().Length() <= squaredDeparture)
```

Comment:
If we compare to the squaredDeparture, then I guess we should also use `ship.Position().LengthSquared()`, otherwise we move quite a distance beyond the intended departure point..

Reply:
done